packages/x-components/src/components/base-extra-params.vue

```diff
@@ -0,0 +1,165 @@
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop, Watch } from 'vue-property-decorator';
+  import { Dictionary } from '../utils';
+  import { ExtraParamXEvent } from '../wiring';
+  import { SnippetConfig } from '../x-installer';
+  import { XInject } from './decorators/injection.decorators';
+
+  /**
+   * It emits a list of {@link ExtraParamXEvent} and the values, which are merged
+   * with the extra params from the snippet config, passed as properties.
+   *
+   * Each extra param is emitted by each event.
+   *
+   * By default if there is an extra param passed as prop, it will override
+   * the snippet config in case it exists.
+   *
+   * @public
+   */
+  @Component
+  export default class BaseExtraParams extends Vue {
+    /**
+     * An array where the values are the {@link ExtraParamXEvent}.
+     * By default this component emits the 'UserChangedRequestParams' event.
+     *
+     * @public
+     */
+    @Prop({ default: () => ['UserChangedRequestParams'] })
+    protected events!: ExtraParamXEvent[];
+
+    /**
+     * (Required) A record where the keys are the extra params and the
+     * value.
+     *
+     * @public
+     */
+    @Prop({ required: true })
+    protected values!: Record<string, any>;
+
+    /**
+     * It injects {@link SnippetConfig} provided by an ancestor.
+     *
+     * @internal
+     */
+    @XInject('snippetConfig')
+    public snippetConfig!: SnippetConfig;
+
+    /**
+     * Computed property that returns the combination of the
+     * extra params from the props and the snippet config.
+     *
+     * @remarks In the merge strategy the props have more priority than the snippet.
+     *
+     * @returns An object with the extra params.
+     *
+     * @internal
+     */
+    protected get extraParams(): Dictionary {
+      const {
+        _instance,
+        _env,
+        _lang,
+        _searchLang,
+        _scope,
+        _consent,
+        _currency,
+        _documentDirection,
+        ...extraParams
+      } = this.snippetConfig ?? {};
+
+      return {
+        ...extraParams,
+        ...this.values
+      };
+    }
+
+    /**
+     * Emits the events passed as prop with the new extra param values.
+     *
+     * @param newExtraParams - New extra params.
+     * @param oldExtraParams - Previous extra params.
+     *
+     * @internal
+     */
+    @Watch('extraParams', { immediate: true, deep: true })
+    protected syncExtraParams(
+      newExtraParams: Record<string, any>,
+      oldExtraParams: Record<string, any>
```

Comment:
As I told you yesterday, this is not a good idea. When watching objects deeply, the watcher might be triggered because the object instance has changed, or because a single property has changed. In this last case, you will [receive in both new and old parameters](https://vuejs.org/v2/api/#vm-watch) the same object with the same values.